lib/pgp.ml

```diff
@@ -537,24 +540,216 @@ module Packet = struct
       decode_convention public_key secret_packet convention
   end
 
+  module Signature = struct
+    module Subpacket = struct
+      type key_flag =
+        | Certification
+        | Sign_data
+        | Encrypt_communication
+        | Encrypt_storage
+        | Split
+        | Authentication
+        | Private_partaged
+        | Unknown_flag
+      [@@deriving ord, eq, show]
+
+      let tag_to_keyflag tag =
+        match tag with
+        | 0 -> Certification
+        | 1 -> Sign_data
+        | 2 -> Encrypt_communication
+        | 3 -> Encrypt_storage
+        | 4 -> Split
+        | 5 -> Authentication
+        | 7 -> Private_partaged
+        | _ -> Unknown_flag
+
+      let keyflag_to_string flag =
+        match flag with
+        | Certification -> "This key may be used to certify other keys."
+        | Sign_data -> "This key may be used to sign data."
+        | Encrypt_communication ->
+          "This key may be used to encrypt communications."
+        | Encrypt_storage -> "This key may be used to encrypt storage."
+        | Split ->
+          "The private component of this key may have been split by a \
+           secret-sharing mechanism."
+        | Authentication -> "This key may be used for authentication."
+        | Private_partaged ->
+          "The private component of this key may be in the possession of more \
+           than one person."
+        | Unknown_flag -> "Unknown key flag"
+
+      type t =
+        | Key_expiration_time of int32
+        | Revocation_reason of string
+        | Issuer_id of string
+        | Uses of key_flag list
+        | Unknown
+      [@@deriving ord, eq, show]
+
+      let get_length cs =
+        let first_octet = Cstruct.get_uint8 cs 0 in
```

Comment:
octet -> byte

Reply:
I agree with this, but the terms first_octet and second_octet are actually used elsewhere in this file, that's file I kept using them. I guess I should change those too.